src/buildApplication/middlewareManifest.ts

```diff
@@ -74,9 +79,12 @@ export function parseMiddlewareManifest(
 	const functionsEntries = Object.values(middlewareManifest.functions);
 
 	for (const [name, filepath] of functionsMap) {
+		// the .vc-config name includes the basePath, so we need to strip it for matching in the middleware manifest.
+		const fileName = (basePath ? name.replace(basePath, '') : name).slice(1);
```

Comment:
maybe we could make this clearer with a regex? what do you think?
```suggestion
		const prefixRegex = new RegExp(`^(${basePath})?/`);
		const fileName = name.replace(basePathRegex, '');
```